core/lib/spree/event/configuration.rb

```diff
@@ -14,12 +18,64 @@ def autoload_subscribers
       end
 
       def adapter
-        @adapter ||= Spree::Event::Adapters::ActiveSupportNotifications
+        @adapter ||= Spree::Core::VersionedValue.new(
+          Spree::Event::Adapters::ActiveSupportNotifications,
+          "4.0.0.alpha" => Spree::Event::Adapters::EventBus.new
```

Comment:
Why one is a class and the other is an instance?

Reply:
One of the main ideas behind building a new adapter is to make it work without relying on global state. Having it as an instance means that we can use its state as storage for the event bus. A clear benefit is that it allows us to use separate buses when testing.

On top of that, even though we're still using a global bus for the whole application to preserve the current API, we could eventually create different buses. For instance:

```ruby
module Spree
  class Order
    include Spree::Event::Adapters::EventBus # we'd want to make it work through `Spree::Event` though

    def method(...)
      fire 'order_finalized', order: self
    end
  end
end
```

Reply:
Got it, so basically the instance of `Spree::Event::Adapters::EventBus.new` has the same interface of the `Spree::Event::Adapters::ActiveSupportNotifications` class/module.

Reply:
Right :slightly_smiling_face: